src/python/WMCore/Services/PyCondor/PyCondorAPI.py

```diff
@@ -53,6 +62,8 @@ def isScheddOverloaded(self):
             logging.exception(msg)
             # since it failed, assume it's overloaded
             isOverloaded = True
+            # if there is an error, try to recreate the collector instance
+            self.coll = htcondor.Collector()
```

Comment:
@goughes, Erik, in this case doesn't need to be retried?

Reply:
Yes, I can update for the collector as well if you think this retry method is ok.

Reply:
Thanks Erik, I think we can do the same thing.